tests/blockchain/test_blockchain.py

```diff
@@ -2633,3 +2633,56 @@ async def test_get_blocks_at(self, empty_blockchain, default_1000_blocks):
         assert blocks
         assert len(blocks) == 200
         assert blocks[-1].height == 199
+
+    @pytest.mark.asyncio
+    @pytest.mark.parametrize("opcode", [ConditionOpcode.ASSERT_SECONDS_RELATIVE, ConditionOpcode.ASSERT_HEIGHT_RELATIVE])
+    @pytest.mark.parametrize("lock_time,expected", [(-1, ReceiveBlockResult.NEW_PEAK), (0, ReceiveBlockResult.NEW_PEAK), (1, ReceiveBlockResult.INVALID_BLOCK)])
+    async def test_ephmeral_timelock(self, empty_blockchain, opcode, lock_time, expected):
+        b = empty_blockchain
+        blocks = bt.get_consecutive_blocks(
+            3,
+            guarantee_transaction_block=True,
+            farmer_reward_puzzle_hash=bt.pool_ph,
+            pool_reward_puzzle_hash=bt.pool_ph,
+        )
+        assert (await b.receive_block(blocks[0]))[0] == ReceiveBlockResult.NEW_PEAK
+        assert (await b.receive_block(blocks[1]))[0] == ReceiveBlockResult.NEW_PEAK
+        assert (await b.receive_block(blocks[2]))[0] == ReceiveBlockResult.NEW_PEAK
+
+        wt: WalletTool = bt.get_pool_wallet_tool()
+
+        conditions = {opcode: [ConditionWithArgs(opcode, [int_to_bytes(lock_time)])]}
+
+        tx1: SpendBundle = wt.generate_signed_transaction(
+            10, wt.get_new_puzzlehash(), list(blocks[-1].get_included_reward_coins())[0],
+            condition_dic=conditions.copy()
+        )
+        coin1: Coin = tx1.additions()[0]
+        tx2: SpendBundle = wt.generate_signed_transaction(
+            10, wt.get_new_puzzlehash(), coin1,
+            condition_dic=conditions.copy()
```

Comment:
One problem is for the `1` case, we're not explicitly checking that it's the ephemeral coin that's failing. It might just be failing due to `coin_0`. Why not just leave the initial `CoinSpend` condition-free? ie. write `condition_dic={}`.

Reply:
maybe this comment got shifted around, but I removed the conditions from the spend of the coinbase, i.e. not this one. Is that what you had in mind?